apps/server/src/adapters/WebsocketAdapter.ts

```diff
@@ -40,70 +41,116 @@ export class SocketServer implements IAdapter {
 
     // eslint-disable-next-line @typescript-eslint/no-this-alias -- this logic is used to ensure singleton
     instance = this;
-    this.clientIds = new Set<string>();
+    this.clients = new Map<string, Client>();
     this.wss = null;
   }
 
   init(server: Server) {
     this.wss = new WebSocketServer({ path: '/ws', server, maxPayload: this.MAX_PAYLOAD });
 
     this.wss.on('connection', (ws) => {
-      let clientId = getRandomName();
-      this.clientIds.add(clientId);
-      logger.info(LogOrigin.Client, `${this.clientIds.size} Connections with new: ${clientId}`);
+      const clientId = generateId();
+
+      this.clients.set(clientId, {
+        type: ClientTypes.Unknown,
+        identify: false,
+        redirect: '',
+        name: getRandomName(),
+        rename: '',
+      });
+      logger.info(LogOrigin.Client, `${this.clients.size} Connections with new: ${clientId}`);
 
-      // send store payload on connect
       ws.send(
         JSON.stringify({
-          type: 'ontime',
-          payload: eventStore.poll(),
+          type: 'client-id',
+          payload: clientId,
         }),
       );
 
       ws.send(
         JSON.stringify({
           type: 'client-name',
-          payload: clientId,
+          payload: this.clients.get(clientId).name,
+        }),
+      );
+
+      this.sendClientList();
+
+      // send store payload on connect
+      ws.send(
+        JSON.stringify({
+          type: 'ontime',
+          payload: eventStore.poll(),
         }),
       );
 
       ws.on('error', console.error);
 
       ws.on('close', () => {
-        this.clientIds.delete(clientId);
-        logger.info(LogOrigin.Client, `${this.clientIds.size} Connections with disconnected: ${clientId}`);
+        this.clients.delete(clientId);
+        logger.info(LogOrigin.Client, `${this.clients.size} Connections with disconnected: ${clientId}`);
+        this.sendClientList();
       });
 
       ws.on('message', (data) => {
         try {
-          // @ts-expect-error -- ??
+          //@ts-expect-error ???
```

Comment:
I am a bit confused about some of the changes here.
I think it is important to keep the changes meaningful